Address warnings in KSCrash fork

https://github.com/ably-forks/KSCrash

Update: I tried merging changes from upstream.
I kept hitting compilation errors like this while running `pod lib lint`

```
KSCrash> pod lib lint

 -> KSCrashAblyFork (1.15.8-ably-1)
    - WARN  | [KSCrashAblyFork/Recording, KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Filters, and more...] xcodebuild:  /Users/cesare/Desktop/KSCrash/Source/KSCrash/Recording/Tools/KSThread.h:74:23: warning: this function declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Recording, KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Filters, and more...] xcodebuild:  /Users/cesare/Desktop/KSCrash/Source/KSCrash/Recording/Tools/KSMachineContext.h:40:29: warning: this function declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Recording, KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Filters, and more...] xcodebuild:  /Users/cesare/Desktop/KSCrash/Source/KSCrash/Recording/Tools/KSMachineContext.h:44:28: warning: this function declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Recording, KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Filters, and more...] xcodebuild:  /Users/cesare/Desktop/KSCrash/Source/KSCrash/Recording/Tools/KSMachineContext.h:64:21: warning: this function declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Recording, KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Filters, and more...] xcodebuild:  /Users/cesare/Desktop/KSCrash/Source/KSCrash/Recording/Tools/KSLogger.h:262:24: warning: this function declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Recording, KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Filters, and more...] xcodebuild:  /Users/cesare/Desktop/KSCrash/Source/KSCrash/Recording/KSCrashC.h:198:27: warning: this function declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Recording, KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Filters, and more...] xcodebuild:  /Users/cesare/Desktop/KSCrash/Source/KSCrash/Recording/KSCrashC.h:229:30: warning: this function declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Recording, KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Filters, and more...] xcodebuild:  /Users/cesare/Desktop/KSCrash/Source/KSCrash/Recording/Monitors/KSCrashMonitor.h:60:42: warning: this function declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Recording, KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Filters, and more...] xcodebuild:  /Users/cesare/Desktop/KSCrash/Source/KSCrash/Recording/Monitors/KSCrashMonitor.h:76:22: warning: this function declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Recording, KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Filters, and more...] xcodebuild:  /Users/cesare/Desktop/KSCrash/Source/KSCrash/Recording/Monitors/KSCrashMonitor_AppState.h:99:24: warning: this function declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Recording, KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Filters, and more...] xcodebuild:  /Users/cesare/Desktop/KSCrash/Source/KSCrash/Recording/Monitors/KSCrashMonitor_AppState.h:128:40: warning: this function declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Recording, KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Filters, and more...] xcodebuild:  /Users/cesare/Desktop/KSCrash/Source/KSCrash/Recording/Monitors/KSCrashMonitor_System.h:40:38: warning: this function declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Recording, KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Filters, and more...] xcodebuild:  /Users/cesare/Desktop/KSCrash/Source/KSCrash/Recording/KSCrashCachedData.h:34:18: warning: this function declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Recording, KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Filters, and more...] xcodebuild:  /Users/cesare/Desktop/KSCrash/Source/KSCrash/Recording/KSCrashCachedData.h:35:20: warning: this function declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Recording, KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Filters, and more...] xcodebuild:  /Users/cesare/Desktop/KSCrash/Source/KSCrash/Recording/KSCrashReportStore.h:55:25: warning: this function declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Recording, KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Filters, and more...] xcodebuild:  /Users/cesare/Desktop/KSCrash/Source/KSCrash/Recording/KSCrashReportStore.h:86:28: warning: this function declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Recording, KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Filters, and more...] xcodebuild:  /Users/cesare/Desktop/KSCrash/Source/KSCrash/Recording/Monitors/KSCrashMonitor_Deadlock.h:53:40: warning: this function declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Recording, KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Filters, and more...] xcodebuild:  /Users/cesare/Desktop/KSCrash/Source/KSCrash/Recording/Monitors/KSCrashMonitor_User.h:68:36: warning: this function declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Recording, KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Filters, and more...] xcodebuild:  /Users/cesare/Desktop/KSCrash/Source/KSCrash/Recording/Monitors/KSCrashMonitor_Zombie.h:62:38: warning: this function declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Recording, KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Filters, and more...] xcodebuild:  /Users/cesare/Desktop/KSCrash/Source/KSCrash/Recording/Monitors/KSCrashMonitor_MachException.h:45:45: warning: this function declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Recording, KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Filters, and more...] xcodebuild:  /Users/cesare/Desktop/KSCrash/Source/KSCrash/Recording/Monitors/KSCrashMonitor_CPPException.h:37:44: warning: this function declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Recording, KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Filters, and more...] xcodebuild:  /Users/cesare/Desktop/KSCrash/Source/KSCrash/Recording/Monitors/KSCrashMonitor_NSException.h:45:43: warning: this function declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Recording, KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Filters, and more...] xcodebuild:  /Users/cesare/Desktop/KSCrash/Source/KSCrash/Recording/Monitors/KSCrashMonitor_Signal.h:45:38: warning: this function declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Recording, KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Filters, and more...] xcodebuild:  /Users/cesare/Desktop/KSCrash/Source/KSCrash/Recording/Monitors/KSCrashMonitor.c:58:33: warning: this function declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Recording, KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Filters, and more...] xcodebuild:  /Users/cesare/Desktop/KSCrash/Source/KSCrash/Recording/Tools/KSDynamicLinker.h:55:20: warning: this function declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Tools, KSCrashAblyFork/Reporting/Sinks, and more...] xcodebuild:  /Users/cesare/Desktop/KSCrash/Source/KSCrash/Reporting/Tools/KSReachabilityKSCrash.h:133:60: warning: this block declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Tools, KSCrashAblyFork/Reporting/Sinks, and more...] xcodebuild:  /Users/cesare/Desktop/KSCrash/Source/KSCrash/Reporting/Tools/KSReachabilityKSCrash.h:149:29: warning: this block declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Tools, KSCrashAblyFork/Reporting/Sinks, and more...] xcodebuild:  /Users/cesare/Desktop/KSCrash/Source/KSCrash/Reporting/Tools/KSReachabilityKSCrash.m:307:60: warning: this block declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Tools, KSCrashAblyFork/Reporting/Sinks, and more...] xcodebuild:  /Users/cesare/Desktop/KSCrash/Source/KSCrash/Reporting/Tools/KSReachabilityKSCrash.m:314:29: warning: this block declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Recording, KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Filters, and more...] xcodebuild:  /Users/cesare/Library/Developer/Xcode/DerivedData/App-eetzhbqgtldrrneakwhizkjdyass/Build/Products/Release-iphonesimulator/KSCrashAblyFork/KSCrashAblyFork.framework/Headers/KSCrashC.h:198:27: warning: this function declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Recording, KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Filters, and more...] xcodebuild:  /Users/cesare/Library/Developer/Xcode/DerivedData/App-eetzhbqgtldrrneakwhizkjdyass/Build/Products/Release-iphonesimulator/KSCrashAblyFork/KSCrashAblyFork.framework/Headers/KSCrashC.h:229:30: warning: this function declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Recording, KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Filters, and more...] xcodebuild:  /Users/cesare/Library/Developer/Xcode/DerivedData/App-eetzhbqgtldrrneakwhizkjdyass/Build/Products/Release-iphonesimulator/KSCrashAblyFork/KSCrashAblyFork.framework/Headers/KSThread.h:74:23: warning: this function declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Recording, KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Filters, and more...] xcodebuild:  /Users/cesare/Library/Developer/Xcode/DerivedData/App-eetzhbqgtldrrneakwhizkjdyass/Build/Products/Release-iphonesimulator/KSCrashAblyFork/KSCrashAblyFork.framework/Headers/KSMachineContext.h:40:29: warning: this function declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Recording, KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Filters, and more...] xcodebuild:  /Users/cesare/Library/Developer/Xcode/DerivedData/App-eetzhbqgtldrrneakwhizkjdyass/Build/Products/Release-iphonesimulator/KSCrashAblyFork/KSCrashAblyFork.framework/Headers/KSMachineContext.h:44:28: warning: this function declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Recording, KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Filters, and more...] xcodebuild:  /Users/cesare/Library/Developer/Xcode/DerivedData/App-eetzhbqgtldrrneakwhizkjdyass/Build/Products/Release-iphonesimulator/KSCrashAblyFork/KSCrashAblyFork.framework/Headers/KSMachineContext.h:64:21: warning: this function declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Recording, KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Filters, and more...] xcodebuild:  /Users/cesare/Library/Developer/Xcode/DerivedData/App-eetzhbqgtldrrneakwhizkjdyass/Build/Products/Release-iphonesimulator/KSCrashAblyFork/KSCrashAblyFork.framework/Headers/KSDynamicLinker.h:55:20: warning: this function declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Recording, KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Filters, and more...] xcodebuild:  /Users/cesare/Library/Developer/Xcode/DerivedData/App-eetzhbqgtldrrneakwhizkjdyass/Build/Products/Release-iphonesimulator/KSCrashAblyFork/KSCrashAblyFork.framework/Headers/KSLogger.h:262:24: warning: this function declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Tools, KSCrashAblyFork/Reporting/Sinks, and more...] xcodebuild:  /Users/cesare/Library/Developer/Xcode/DerivedData/App-eetzhbqgtldrrneakwhizkjdyass/Build/Products/Release-iphonesimulator/KSCrashAblyFork/KSCrashAblyFork.framework/Headers/KSReachabilityKSCrash.h:133:60: warning: this block declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Tools, KSCrashAblyFork/Reporting/Sinks, and more...] xcodebuild:  /Users/cesare/Library/Developer/Xcode/DerivedData/App-eetzhbqgtldrrneakwhizkjdyass/Build/Products/Release-iphonesimulator/KSCrashAblyFork/KSCrashAblyFork.framework/Headers/KSReachabilityKSCrash.h:149:29: warning: this block declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Recording, KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Filters, and more...] xcodebuild:  /Users/cesare/Desktop/KSCrash/Source/KSCrash/Recording/Monitors/KSCrashMonitor_System.m:502:72: warning: 'operatingSystemVersion' is only available on macOS 10.10 or newer [-Wunguarded-availability]
    - NOTE  | [KSCrashAblyFork/Recording, KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Filters, and more...] xcodebuild:  /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.13.sdk/System/Library/Frameworks/Foundation.framework/Headers/NSProcessInfo.h:54:47: note: 'operatingSystemVersion' has been explicitly marked partial here
    - NOTE  | [KSCrashAblyFork/Recording, KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Filters, and more...] xcodebuild:  /Users/cesare/Desktop/KSCrash/Source/KSCrash/Recording/Monitors/KSCrashMonitor_System.m:502:72: note: enclose 'operatingSystemVersion' in an @available check to silence this warning
    - WARN  | [KSCrashAblyFork/Recording, KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Filters, and more...] xcodebuild:  /Users/cesare/Library/Developer/Xcode/DerivedData/App-eetzhbqgtldrrneakwhizkjdyass/Build/Products/Release/KSCrashAblyFork/KSCrashAblyFork.framework/Headers/KSCrashC.h:198:27: warning: this function declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Recording, KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Filters, and more...] xcodebuild:  /Users/cesare/Library/Developer/Xcode/DerivedData/App-eetzhbqgtldrrneakwhizkjdyass/Build/Products/Release/KSCrashAblyFork/KSCrashAblyFork.framework/Headers/KSCrashC.h:229:30: warning: this function declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Recording, KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Filters, and more...] xcodebuild:  /Users/cesare/Library/Developer/Xcode/DerivedData/App-eetzhbqgtldrrneakwhizkjdyass/Build/Products/Release/KSCrashAblyFork/KSCrashAblyFork.framework/Headers/KSThread.h:74:23: warning: this function declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Recording, KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Filters, and more...] xcodebuild:  /Users/cesare/Library/Developer/Xcode/DerivedData/App-eetzhbqgtldrrneakwhizkjdyass/Build/Products/Release/KSCrashAblyFork/KSCrashAblyFork.framework/Headers/KSMachineContext.h:40:29: warning: this function declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Recording, KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Filters, and more...] xcodebuild:  /Users/cesare/Library/Developer/Xcode/DerivedData/App-eetzhbqgtldrrneakwhizkjdyass/Build/Products/Release/KSCrashAblyFork/KSCrashAblyFork.framework/Headers/KSMachineContext.h:44:28: warning: this function declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Recording, KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Filters, and more...] xcodebuild:  /Users/cesare/Library/Developer/Xcode/DerivedData/App-eetzhbqgtldrrneakwhizkjdyass/Build/Products/Release/KSCrashAblyFork/KSCrashAblyFork.framework/Headers/KSMachineContext.h:64:21: warning: this function declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Recording, KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Filters, and more...] xcodebuild:  /Users/cesare/Library/Developer/Xcode/DerivedData/App-eetzhbqgtldrrneakwhizkjdyass/Build/Products/Release/KSCrashAblyFork/KSCrashAblyFork.framework/Headers/KSDynamicLinker.h:55:20: warning: this function declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Recording, KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Filters, and more...] xcodebuild:  /Users/cesare/Library/Developer/Xcode/DerivedData/App-eetzhbqgtldrrneakwhizkjdyass/Build/Products/Release/KSCrashAblyFork/KSCrashAblyFork.framework/Headers/KSLogger.h:262:24: warning: this function declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Tools, KSCrashAblyFork/Reporting/Sinks, and more...] xcodebuild:  /Users/cesare/Library/Developer/Xcode/DerivedData/App-eetzhbqgtldrrneakwhizkjdyass/Build/Products/Release/KSCrashAblyFork/KSCrashAblyFork.framework/Headers/KSReachabilityKSCrash.h:133:60: warning: this block declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Tools, KSCrashAblyFork/Reporting/Sinks, and more...] xcodebuild:  /Users/cesare/Library/Developer/Xcode/DerivedData/App-eetzhbqgtldrrneakwhizkjdyass/Build/Products/Release/KSCrashAblyFork/KSCrashAblyFork.framework/Headers/KSReachabilityKSCrash.h:149:29: warning: this block declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Recording, KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Filters, and more...] xcodebuild:  /Users/cesare/Library/Developer/Xcode/DerivedData/App-eetzhbqgtldrrneakwhizkjdyass/Build/Products/Release-appletvsimulator/KSCrashAblyFork/KSCrashAblyFork.framework/Headers/KSCrashC.h:198:27: warning: this function declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Recording, KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Filters, and more...] xcodebuild:  /Users/cesare/Library/Developer/Xcode/DerivedData/App-eetzhbqgtldrrneakwhizkjdyass/Build/Products/Release-appletvsimulator/KSCrashAblyFork/KSCrashAblyFork.framework/Headers/KSCrashC.h:229:30: warning: this function declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Recording, KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Filters, and more...] xcodebuild:  /Users/cesare/Library/Developer/Xcode/DerivedData/App-eetzhbqgtldrrneakwhizkjdyass/Build/Products/Release-appletvsimulator/KSCrashAblyFork/KSCrashAblyFork.framework/Headers/KSThread.h:74:23: warning: this function declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Recording, KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Filters, and more...] xcodebuild:  /Users/cesare/Library/Developer/Xcode/DerivedData/App-eetzhbqgtldrrneakwhizkjdyass/Build/Products/Release-appletvsimulator/KSCrashAblyFork/KSCrashAblyFork.framework/Headers/KSMachineContext.h:40:29: warning: this function declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Recording, KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Filters, and more...] xcodebuild:  /Users/cesare/Library/Developer/Xcode/DerivedData/App-eetzhbqgtldrrneakwhizkjdyass/Build/Products/Release-appletvsimulator/KSCrashAblyFork/KSCrashAblyFork.framework/Headers/KSMachineContext.h:44:28: warning: this function declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Recording, KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Filters, and more...] xcodebuild:  /Users/cesare/Library/Developer/Xcode/DerivedData/App-eetzhbqgtldrrneakwhizkjdyass/Build/Products/Release-appletvsimulator/KSCrashAblyFork/KSCrashAblyFork.framework/Headers/KSMachineContext.h:64:21: warning: this function declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Recording, KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Filters, and more...] xcodebuild:  /Users/cesare/Library/Developer/Xcode/DerivedData/App-eetzhbqgtldrrneakwhizkjdyass/Build/Products/Release-appletvsimulator/KSCrashAblyFork/KSCrashAblyFork.framework/Headers/KSDynamicLinker.h:55:20: warning: this function declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Recording, KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Filters, and more...] xcodebuild:  /Users/cesare/Library/Developer/Xcode/DerivedData/App-eetzhbqgtldrrneakwhizkjdyass/Build/Products/Release-appletvsimulator/KSCrashAblyFork/KSCrashAblyFork.framework/Headers/KSLogger.h:262:24: warning: this function declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Tools, KSCrashAblyFork/Reporting/Sinks, and more...] xcodebuild:  /Users/cesare/Library/Developer/Xcode/DerivedData/App-eetzhbqgtldrrneakwhizkjdyass/Build/Products/Release-appletvsimulator/KSCrashAblyFork/KSCrashAblyFork.framework/Headers/KSReachabilityKSCrash.h:133:60: warning: this block declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Tools, KSCrashAblyFork/Reporting/Sinks, and more...] xcodebuild:  /Users/cesare/Library/Developer/Xcode/DerivedData/App-eetzhbqgtldrrneakwhizkjdyass/Build/Products/Release-appletvsimulator/KSCrashAblyFork/KSCrashAblyFork.framework/Headers/KSReachabilityKSCrash.h:149:29: warning: this block declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Tools, KSCrashAblyFork/Reporting/Sinks, and more...] xcodebuild:  /Users/cesare/Desktop/KSCrash/Source/KSCrash/Reporting/Tools/KSReachabilityKSCrash.m:370:60: warning: this block declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Tools, KSCrashAblyFork/Reporting/Sinks, and more...] xcodebuild:  /Users/cesare/Desktop/KSCrash/Source/KSCrash/Reporting/Tools/KSReachabilityKSCrash.m:377:29: warning: this block declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Recording, KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Filters, and more...] xcodebuild:  /Users/cesare/Library/Developer/Xcode/DerivedData/App-eetzhbqgtldrrneakwhizkjdyass/Build/Products/Release-watchsimulator/KSCrashAblyFork/KSCrashAblyFork.framework/Headers/KSCrashC.h:198:27: warning: this function declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Recording, KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Filters, and more...] xcodebuild:  /Users/cesare/Library/Developer/Xcode/DerivedData/App-eetzhbqgtldrrneakwhizkjdyass/Build/Products/Release-watchsimulator/KSCrashAblyFork/KSCrashAblyFork.framework/Headers/KSCrashC.h:229:30: warning: this function declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Recording, KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Filters, and more...] xcodebuild:  /Users/cesare/Library/Developer/Xcode/DerivedData/App-eetzhbqgtldrrneakwhizkjdyass/Build/Products/Release-watchsimulator/KSCrashAblyFork/KSCrashAblyFork.framework/Headers/KSThread.h:74:23: warning: this function declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Recording, KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Filters, and more...] xcodebuild:  /Users/cesare/Library/Developer/Xcode/DerivedData/App-eetzhbqgtldrrneakwhizkjdyass/Build/Products/Release-watchsimulator/KSCrashAblyFork/KSCrashAblyFork.framework/Headers/KSMachineContext.h:40:29: warning: this function declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Recording, KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Filters, and more...] xcodebuild:  /Users/cesare/Library/Developer/Xcode/DerivedData/App-eetzhbqgtldrrneakwhizkjdyass/Build/Products/Release-watchsimulator/KSCrashAblyFork/KSCrashAblyFork.framework/Headers/KSMachineContext.h:44:28: warning: this function declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Recording, KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Filters, and more...] xcodebuild:  /Users/cesare/Library/Developer/Xcode/DerivedData/App-eetzhbqgtldrrneakwhizkjdyass/Build/Products/Release-watchsimulator/KSCrashAblyFork/KSCrashAblyFork.framework/Headers/KSMachineContext.h:64:21: warning: this function declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Recording, KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Filters, and more...] xcodebuild:  /Users/cesare/Library/Developer/Xcode/DerivedData/App-eetzhbqgtldrrneakwhizkjdyass/Build/Products/Release-watchsimulator/KSCrashAblyFork/KSCrashAblyFork.framework/Headers/KSDynamicLinker.h:55:20: warning: this function declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Recording, KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Filters, and more...] xcodebuild:  /Users/cesare/Library/Developer/Xcode/DerivedData/App-eetzhbqgtldrrneakwhizkjdyass/Build/Products/Release-watchsimulator/KSCrashAblyFork/KSCrashAblyFork.framework/Headers/KSLogger.h:262:24: warning: this function declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Tools, KSCrashAblyFork/Reporting/Sinks, and more...] xcodebuild:  /Users/cesare/Library/Developer/Xcode/DerivedData/App-eetzhbqgtldrrneakwhizkjdyass/Build/Products/Release-watchsimulator/KSCrashAblyFork/KSCrashAblyFork.framework/Headers/KSReachabilityKSCrash.h:133:60: warning: this block declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | [KSCrashAblyFork/Reporting, KSCrashAblyFork/Reporting/Tools, KSCrashAblyFork/Reporting/Sinks, and more...] xcodebuild:  /Users/cesare/Library/Developer/Xcode/DerivedData/App-eetzhbqgtldrrneakwhizkjdyass/Build/Products/Release-watchsimulator/KSCrashAblyFork/KSCrashAblyFork.framework/Headers/KSReachabilityKSCrash.h:149:29: warning: this block declaration is not a prototype [-Wstrict-prototypes]
    - ERROR | [iOS] [KSCrashAblyFork/Reporting/Filters/Alert] unknown: Encountered an unknown error (pathname contains null byte) during validation.
```

Same kind of errors (but fever) happen if I lint the [upstream master](https://github.com/kstenerud/KSCrash)

```
 -> KSCrash (1.15.16)
    - ERROR | [KSCrash/Recording, KSCrash/Reporting, KSCrash/Reporting/Filters, and more...] xcodebuild: Returned an unsuccessful exit code. You can use `--verbose` for more information.
    - ERROR | [KSCrash/Recording, KSCrash/Reporting, KSCrash/Reporting/Filters, and more...] xcodebuild:  /Users/cesare/Desktop/aaa/KSCrash-master/Source/KSCrash/Recording/Monitors/KSCrashMonitor_Signal.c:202:5: error: use of undeclared identifier 'g_signalStack'
```

--- a/Ably.podspec
+++ b/Ably.podspec
@@ -19,5 +19,5 @@ Pod::Spec.new do |s|
   s.module_map        = 'Source/Ably.modulemap'
   s.dependency 'SocketRocket', '0.5.1'
   s.dependency 'msgpack', '0.1.8'
-  s.dependency 'KSCrashAblyFork', '1.15.8-ably-1'
+  s.dependency 'KSCrashAblyFork', '1.15.8-ably-2'
 end

--- a/Cartfile
+++ b/Cartfile
@@ -1,3 +1,3 @@
 github "facebook/SocketRocket" "28035e1a98a427853e4038ff1b70479fa8374cfa"
 github "rvi/msgpack-objective-C" "ad0a627becfd27ca45dfa6684b3fc79f058f5ce9"
-github "ably-forks/KSCrash" "639c68762e1ad62d52c89eac1c23178f76ce69ef"
+github "ably-forks/KSCrash" "1.15.8-ably-2"

--- a/Cartfile
+++ b/Cartfile
@@ -1,3 +1,3 @@
 github "facebook/SocketRocket" "28035e1a98a427853e4038ff1b70479fa8374cfa"
-github "rvi/msgpack-objective-C" "master"
+github "rvi/msgpack-objective-C" "ad0a627becfd27ca45dfa6684b3fc79f058f5ce9"
 github "ably-forks/KSCrash" "639c68762e1ad62d52c89eac1c23178f76ce69ef"

--- a/Cartfile
+++ b/Cartfile
@@ -1,3 +1,3 @@
 github "facebook/SocketRocket" "28035e1a98a427853e4038ff1b70479fa8374cfa"
 github "rvi/msgpack-objective-C" "master"
-github "ably-forks/KSCrash" "1.15.8-ably-1"
+github "ably-forks/KSCrash" "639c68762e1ad62d52c89eac1c23178f76ce69ef"

--- a/Cartfile.resolved
+++ b/Cartfile.resolved
@@ -1,3 +1,3 @@
-github "ably-forks/KSCrash" "1.15.8-ably-1"
+github "ably-forks/KSCrash" "639c68762e1ad62d52c89eac1c23178f76ce69ef"
 github "facebook/SocketRocket" "28035e1a98a427853e4038ff1b70479fa8374cfa"
 github "rvi/msgpack-objective-C" "ad0a627becfd27ca45dfa6684b3fc79f058f5ce9"

--- a/Cartfile.resolved
+++ b/Cartfile.resolved
@@ -1,3 +1,3 @@
-github "ably-forks/KSCrash" "639c68762e1ad62d52c89eac1c23178f76ce69ef"
+github "ably-forks/KSCrash" "1.15.8-ably-2"
 github "facebook/SocketRocket" "28035e1a98a427853e4038ff1b70479fa8374cfa"
 github "rvi/msgpack-objective-C" "ad0a627becfd27ca45dfa6684b3fc79f058f5ce9"

